paddlepaddle/paddlepaddle-model-zoo/src/test/java/ai/djl/paddlepaddle/zoo/OCRTest.java

```diff
@@ -0,0 +1,276 @@
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package ai.djl.paddlepaddle.zoo;
+
+import ai.djl.Application;
+import ai.djl.MalformedModelException;
+import ai.djl.Model;
+import ai.djl.ModelException;
+import ai.djl.inference.Predictor;
+import ai.djl.modality.cv.Image;
+import ai.djl.modality.cv.ImageFactory;
+import ai.djl.modality.cv.output.BoundingBox;
+import ai.djl.modality.cv.output.Rectangle;
+import ai.djl.modality.cv.util.NDImageUtils;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.repository.zoo.ModelZoo;
+import ai.djl.repository.zoo.ZooModel;
+import ai.djl.translate.Batchifier;
+import ai.djl.translate.TranslateException;
+import ai.djl.translate.Translator;
+import ai.djl.translate.TranslatorContext;
+import ai.djl.util.Utils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.testng.annotations.Test;
+
+public class OCRTest {
+
+    @Test
+    public void testOCR() throws IOException, ModelException, TranslateException {
+        String url =
+                "https://raw.githubusercontent.com/PaddlePaddle/PaddleOCR/dygraph/doc/imgs/11.jpg";
+        Image img = ImageFactory.getInstance().fromUrl(url);
+        List<BoundingBox> boxes = detectWords(img);
+
+        // load Character model
+        Predictor<Image, String> recognizer = getRecognizer();
+        Predictor<Image, Image> rotator = getRotateClassifer();
+        for (int i = 0; i < boxes.size(); i++) {
+            Image subImg = getSubImage(img, boxes.get(i));
+            subImg = rotator.predict(subImg);
+            String name = recognizer.predict(subImg);
+            System.out.println(name);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static List<BoundingBox> detectWords(Image img)
+            throws ModelException, IOException, TranslateException {
+        Class<List<BoundingBox>> clazz = (Class) List.class;
+        Criteria<Image, List<BoundingBox>> criteria =
+                Criteria.builder()
+                        .optApplication(Application.CV.OBJECT_DETECTION)
+                        .optEngine("PaddlePaddle")
+                        .setTypes(Image.class, clazz)
+                        .optModelUrls(
+                                "https://alpha-djl-demos.s3.amazonaws.com/model/paddleOCR/mobile/det_db.zip")
```

Comment:
Let put in our production s3: s3://resources.djl.ai/test-models/test-models/paddleOCR/det_db.zip